_aggregations/bucket/date-histogram.md

```diff
@@ -21,50 +34,265 @@ GET opensearch_dashboards_sample_data_logs/_search
+
+```json
+GET my-logs/_search
+{
+  "size": 0,
+  "aggs": {
+    "by_hour_script": {
+      "date_histogram": {
+        "script": {
+          "lang": "painless",
+          "source": "return doc['timestamp'].value.toInstant().toEpochMilli();"
+        },
+        "fixed_interval": "1h"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
 
-The response has three months worth of logs. If you graph these values, you can see the peak and valleys of the request traffic to your website month over month.
+## Parameters
 
-### Parameters
+The `date_histogram` supports the following parameters.
 
-`date_histogram` aggregations support the following parameters.
+| Parameter | Required | Type | Description |
+|:--|:--|:--|:--|
+| `field` | One of the following is required: `field` or `script` | String | The date/datetime field to bucket on. |
+| `calendar_interval` | One of the following is required: `calendar_interval`, `fixed_interval` or legacy `interval` | String | The calendar-aware interval (e.g., `"day"`, `"1M"`, `"year"`). Only singular calendar units are supported. |
+| `fixed_interval` | One of the following is required: `calendar_interval`, `fixed_interval` or legacy `interval` | String | The exact-duration interval, for example: `"5m"`, `"12h"`, `"30d"`. Not for calendar units like months or quarters. |
+| `time_zone` | Optional | String | The time zone used for bucketing and formatting. Accepts timezone, such as `"Europe/Dublin"` or UTC offsets, such as `"-07:00"`. |
+| `format` | Optional | String | The output date format used for `key_as_string`, for example, `"yyyy-MM-dd"`. If omitted, mapping defaults apply. |
+| `offset` | Optional | String | Shifts bucket boundaries by a positive or negative duration, for example, `"+6h"`, `"-30m"`. Calculated after `time_zone` is applied. |
+| `min_doc_count` | Optional | Integer | The minimum number of docs required to return a bucket. Default is `1`. Set to `0` to include empty buckets. |
+| `extended_bounds` | Optional | Object | Extends the range of buckets beyond your data: `{"min": "<date>", "max": "<date>"}`. Often used with `min_doc_count: 0`. |
+| `hard_bounds` | Optional | Object | Strictly limits buckets to a range: `{"min": "<date>", "max": "<date>"}`. Buckets outside the range are never created. |
+| `missing` | Optional | Date string | Treat docs missing the field as if they had this date value. |
+| `keyed` | Optional | Boolean | When `true`, returns buckets as an object keyed by the formatted date string. |
+| `order` | Optional | Object | Sort buckets by `_key` or `_count`, ascending or descending. |
+| `script` | One of the following is required: `field` or `script` | Object | Optional script to compute the value to bucket on. Use with care; scripts add overhead. |
```

Comment:
Optional script to compute the value to bucket on. Since the scripts are operated to modify each value, they add overhead and should be used cautiously.